### DISCUSSION TITLE
Unique Name Validation

### DISCUSSION THREAD

**Discussion by kapishsingla** [eclipse-langium/langium#1750] [Q&A]
Hi,

I'm implementing validation to ensure each entity has a unique name across file boundaries. Each file corresponds to a Langium document, and each document contains a single entity.

<img width="277" alt="image" src="https://github.com/user-attachments/assets/ca3bf320-bb76-4d29-a65d-e674250c6966">

For Instance,
file1 a.ext -> entity abc{}
file2 b.ext -> entity abc{}

We need to validate that Entity's name should not be same across the different file.

Request to help us with an optimized approach?

**Comment by ydaveluy** [comment]
Hi,

You can use the IndexManager to retrieve all entities and cache the result in a multimap.
This project might help:
https://github.com/ydaveluy/xsmp-modeler/blob/039364476f5b2a6ec023d4edf4104c1794f2d534/src/language/validation/xsmpproject-validator.ts#L81

I don't know if there is a more optimized approach.
